R/funs-string.R

```diff
@@ -318,3 +318,13 @@ pl_word_stringr <- function(string, start = 1L, end = start, sep = " ", ...) {
   check_empty_dots(...)
   string$str$split(sep)$list$gather((start:end) - 1L)$list$join(sep)
 }
+
+pl_str_replace_na_stringr <- function(string, replacement = "NA", ...) {
+  check_empty_dots(...)
+  string$str$replace_strict(
+    old = NA, 
+    new = replacement,
+    default = string$str,
+    return_dtype = pl$String
+  )
```

Comment:
When this is reached, `string` has already been converted to a Polars format:

* if it was a function, say `x`, then here it has the value `pl$col("x")`
* if it was a character vector, say `c("a", "b")` then here it has the value `pl$lit(c("a", "b"))`

This is why we can chain functions using `$`. However, how you pointed out, `replace_strict()` doesn't belong to the `str` namespace, so you can replace `string$str$replace_strict()` by `string$replace_strict()`.

Similarly, you can replace `default = string$str` by `default = string`.

Let me know if that works for you.

Reply:
A second thing to fix here (less important) is that you don't need to specify `...` here because `str_replace_na()` only takes two arguments and we cover them (https://stringr.tidyverse.org/reference/str_replace_na.html).

`...` is needed when some arguments exist in the original function but are not used internally by our Polars translation (either because Polars simply don't have equivalent or because they're tricky to translate).

Reply:
Yeah that wasn't far. I'm used to the internals but I get it's not easy to wrap your head around them. In addition to tests, can you also:

* update the list of supported functions in `vignettes/r-and-polars-expressions.Rmd`
* add a bullet point in the NEWS with your name/github handle (if you want) + the PR number

?